### PR TITLE
LocalRepository UnitOfWork Listener Not Called Upon Successful Operation

### PR DESCRIPTION
**Pull request by elvisisking**
Fixed a problem where the callback listener was not being called when the operation was successful.

### CODE REVIEW

File: plugins/org.komodo.core/src/org/komodo/repository/LocalRepository.java [hunk 1]
```diff
@@ -308,18 +308,27 @@ public void respond( final Object results ) {
...
 
                 if (noTimeout) {
-                    callback.respond(null);
+                    if (getCallback() != null) {
+                        getCallback().respond( null );
+                    }
                 } else {
-                    callback.errorOccurred(new KException(Messages.getString(Messages.LocalRepository.Commit_Timeout, getName())));
+                    if (error == null) {
+                        callback.errorOccurred( new KException( Messages.getString( Messages.LocalRepository.Commit_Timeout,
```
Comment (phantomjinx) [inline plugins/org.komodo.core/src/org/komodo/repository/LocalRepository.java]: What if callback is null here?

File: plugins/org.komodo.core/src/org/komodo/repository/LocalRepository.java [hunk 2]
```diff
@@ -358,18 +367,27 @@ public void respond( final Object results ) {
...
 
             if (noTimeout) {
-                callback.respond(null);
+                if (getCallback() != null) {
+                    getCallback().respond( null );
+                }
             } else {
-                callback.errorOccurred(new KException(Messages.getString(Messages.LocalRepository.Rollback_Timeout, getName())));
+                if (error == null) {
+                    callback.errorOccurred( new KException( Messages.getString( Messages.LocalRepository.Rollback_Timeout,
```
Comment (phantomjinx) [inline plugins/org.komodo.core/src/org/komodo/repository/LocalRepository.java]: ditto here

File: plugins/org.komodo.core/src/org/komodo/repository/LocalRepository.java (hunk 1 above)
Comment (elvisisking) [inline plugins/org.komodo.core/src/org/komodo/repository/LocalRepository.java]: I don't think `callback` can ever be `null` here as it is constructed above in line 306. Now `getCallback()` can be `null`. Naming is a bit confusing as that method does not return the `callback` instance but returns the listener passed in when the `UnitOfWork` was constructed. Am I looking at this right?

Comment (phantomjinx) [inline plugins/org.komodo.core/src/org/komodo/repository/LocalRepository.java]: Ok

**Pull request merged**